### ISSUE TITLE
Chap. 6.3, recurrent dropout, cannot reproduce book's results with TF2.9.1, Keras2.9.0

### ISSUE THREAD

**Issue by fangxuehouwuming** [keras-team/keras#16932]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras):  No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):   Win10
- TensorFlow installed from (source or binary):   Anaconda
- TensorFlow version (use command below):  2.9.1
- Python version:    3.9
- GPU model and memory:  None

**Describe the problem**.
I am going through Francois Chollet's book "Deep Learning with Python" and running the code in his Jupyter Notebooks with Keras: 2.9.0 , Tensorflow: 2.9.1. 
Notebook 6.3, (under the heading "1.6 Using recurrent dropout to fight overfitting") has a model following:

```python
model = Sequential()
model.add(layers.GRU(32, 
                     dropout=0.2,
                     recurrent_dropout=0.2,
                     input_shape=(None, float_data.shape[-1])))
model.add(layers.Dense(1))

model.compile(optimizer=RMSprop(), loss='mae')
history = model.fit_generator(train_gen,
                              steps_per_epoch=500,  
                              epochs=40,
                              validation_data=val_gen,
                              validation_steps=val_steps)
```

Running parts 6.3.6 (recurrent dropout) and 6.3.7 (stacked recurrent layers)  gives the following training and validation losses:

6.3.6
![1661232542955](https://user-images.githubusercontent.com/74046524/186118937-0f7ccf47-c025-4530-a8e0-e1e145f9e02f.png)

6.3.7
![1661246457143](https://user-images.githubusercontent.com/74046524/186122056-82d61282-e6b9-4cd7-b822-9e52fb296c5c.png)

**Describe the expected behavior**.
After using dropout it should solve the overfitting problem just like in the book.

**Comment by sushreebarsa** [comment]
@fangxuehouwuming 
In order to expedite the trouble-shooting process, please provide a  complete code snippet to reproduce the issue reported here. Thank you!

**Comment by fangxuehouwuming** [comment]
+ The corresponding notebook for the code is [deep-learning-with-python-notebooks/first_edition/6.3-advanced-usage-of-recurrent-neural-networks.ipynb](https://github.com/fchollet/deep-learning-with-python-notebooks/blob/master/first_edition/6.3-advanced-usage-of-recurrent-neural-networks.ipynb)

+ The following is the concrete code fragments that related to my question. I'm pretty much following the book.

+ I've already packaged the dataset:
[jena_climate_2009_2016.zip](https://github.com/keras-team/keras/files/9420822/jena_climate_2009_2016.zip)


```python
from keras.optimizers import RMSprop
from keras import layers
from keras.models import Sequential
import matplotlib.pyplot as plt
import numpy as np
import os

# %%
data_dir = r'D:\Study\Python\原始数据集\jena_climate'
fname = os.path.join(data_dir, 'jena_climate_2009_2016.csv')

f = open(fname)
data = f.read()
f.close()

lines = data.split('\n')
header = lines[0].split(',')
lines = lines[1:]
lines.pop()

float_data = np.zeros((len(lines), len(header)-1))
for i, line in enumerate(lines):
    values = [float(x) for x in line.split(',')[1:]]
    float_data[i] = values

# %%
mean = float_data[:200000].mean(axis=0)
std = float_data[:200000].std(axis=0)
float_data = (float_data-mean)/std

def generator(data, lookback, delay, min_index, max_index,
              shuffle=False, batch_size=128, step=6):
    if max_index is None:
        max_index = len(data) - delay - 1
    i = min_index + lookback
    while 1:
        if shuffle:
            rows = np.random.randint(
                min_index + lookback, max_index, size=batch_size)
        else:
            if i + batch_size >= max_index:
                i = min_index + lookback
            rows = np.arange(i, min(i + batch_size, max_index))
            i += len(rows)

        samples = np.zeros((len(rows),
                           lookback // step,
                           data.shape[-1]))
        targets = np.zeros((len(rows),))
        for j, row in enumerate(rows):
            indices = range(rows[j] - lookback, rows[j], step)
            samples[j] = data[indices]
            targets[j] = data[rows[j] + delay][1]
        yield samples, targets

# %%
lookback = 1440
step = 6
delay = 144
batch_size = 128

train_gen = generator(float_data,
                      lookback=lookback,
                      delay=delay,
                      min_index=0,
                      max_index=200000,
                      shuffle=True,
                      step=step,
                      batch_size=batch_size)
val_gen = generator(float_data,
                    lookback=lookback,
                    delay=delay,
                    min_index=200001,
                    max_index=300000,
                    step=step,
                    batch_size=batch_size)
test_gen = generator(float_data,
                     lookback=lookback,
                     delay=delay,
                     min_index=300001,
                     max_index=None,
                     step=step,
                     batch_size=batch_size)

val_steps = (300000 - 200001 - lookback) // batch_size
test_steps = (len(float_data) - 300001 - lookback) // batch_size

# %% GRU with dropout
model = Sequential()
model.add(layers.GRU(32,
                     dropout=0.2,
                     recurrent_dropout=0.2,
                     input_shape=(None, float_data.shape[-1])))
model.add(layers.Dense(1))

model.compile(optimizer=RMSprop(), loss='mae')
history=model.fit_generator(train_gen,
                              steps_per_epoch=500,
                              epochs=40,
                              validation_data=val_gen,
                              validation_steps=val_steps)

# %%
loss = history.history['loss']
val_loss = history.history['val_loss']
epochs = range(1, len(loss)+1)
plt.plot(epochs, loss, 'bo', label='Training loss')
plt.plot(epochs, val_loss, 'b', label='Validation loss')
plt.xlabel('Epochs')
plt.ylabel('Loss')
plt.title('Training and validation loss')
plt.legend()
plt.show()
```

**Comment by sushreebarsa** [comment]
@gowthamkpr I was able to replicate the issue on colab, please find the [gist](https://colab.research.google.com/gist/sushreebarsa/660c00f033c3a00d7fd27ecc5af6751e/untitled2.ipynb) here. Thank you!
